wallet/createtx.go

```diff
@@ -446,3 +429,76 @@ func validateMsgTx(tx *wire.MsgTx, prevScripts [][]byte,
 	}
 	return nil
 }
+
+// SelectableCoin is an interface that represents a coin that can be selected
+// for coin selection.
+type SelectableCoin interface {
+	// Output returns the transaction output of the coin.
+	Output() *wire.TxOut
+}
+
+// SelectableCredit is a wrapper around a wtxmgr.Credit that implements the
+// SelectableCoin interface.
+type SelectableCredit struct {
+	*wtxmgr.Credit
+}
+
+// Output returns the transaction output of the credit.
+func (s *SelectableCredit) Output() *wire.TxOut {
+	return &wire.TxOut{
+		Value:    int64(s.Amount),
+		PkScript: s.PkScript,
+	}
+}
+
+// sortByAmount is a generic sortable type for sorting coins by their amount.
+type sortByAmount[T SelectableCoin] []T
+
+func (s sortByAmount[T]) Len() int { return len(s) }
+func (s sortByAmount[T]) Less(i, j int) bool {
+	return s[i].Output().Value < s[j].Output().Value
+}
+func (s sortByAmount[T]) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// ArrangeCoins takes a list of coins and arranges them according to the
+// specified coin selection strategy and fee rate.
+func ArrangeCoins[T SelectableCoin](coinSelectionStrategy CoinSelectionStrategy,
```

Comment:
Type param usage doesn't feel absolutely necessary here? At least in the context of the wallet today, seems like we could jut swap in `SelectableCoin` as the interface are here instead of `T`. I think reaching for the type param here would make more sense if we already had several other instances of something like `ArrangeCoins` that accepted a diff concrete type. 

Will take a look at the other dependent PRs here as well. Not a blocker, just mainly curious.

Reply:
The reason for using the type parameter here is because Golang doesn't allow using a struct slice when an interface slice is expected.
For example, if `ArrangeCoins` took `eligigle []SelectableCoin`, we couldn't pass in a slice of `[]*SelectableCredit`.
Not a big deal here since we're wrapping the `wtxmgr.Credit` anyway.
But in `lnd` we could pass in `chanfunding.Coin` directly if we use the typed parameter.
But wrapping that wouldn't be a big deal either, so I can go with the interface if you prefer.

Reply:
I resolved this by just moving the `chanfunding.Coin` declaration into the wallet. I think it makes sense to have a specific type that contains just exactly what's needed for coin selection.